classes/classes/Output.as

```diff
@@ -0,0 +1,151 @@
+		{
+			return _addText(text);
+		}
+
+		/**
+		 * Add markdown formatted text (NYI!) to the output-buffer
+		 *
+		 * Actually this is a wrapper around _addText(text, true)
+		 * Unfortunately no one succeded to support markdown formatting for CoC, so as of speaking, this does exactly the same, as text("...")
+		 *
+		 * @param   text    The text to be formatted with markdown (NYI!)
+		 * @return  The instance of the class to support the 'Fluent interface' aka method-chaining
+		 */
+		public function markdown(text:String):Output
+		{
+			return _addText(text, true);
+		}
+
+		/**
+		 * Flushes the buffered output to the GUI aka displaying it
+		 *
+		 * This doesn't clear the output buffer, so you can add more text after that and flush it again.
+		 * flush() always ends a method chain, so you need to start a new one.
+		 */
+		public function flush():void
+		{
+			var fmt:TextFormat = mainView.mainText.getTextFormat();
+
+			if (flags[kFLAGS.CUSTOM_FONT_SIZE] != 0)
+				fmt.size = flags[kFLAGS.CUSTOM_FONT_SIZE];
+
+			mainView.setOutputText(_currentText);
+
+			if (flags[kFLAGS.CUSTOM_FONT_SIZE] != 0)
+				mainView.mainText.setTextFormat(fmt);
+
+			if (mainViewManager.mainColorArray[flags[kFLAGS.BACKGROUND_STYLE]] != null)
+				mainView.mainText.textColor = mainViewManager.mainColorArray[flags[kFLAGS.BACKGROUND_STYLE]];
+		}
+
+		/**
+		 * Adds a formatted headline to the output-buffer
+		 *
+		 * @param	headLine    The headline to be formatted and added
+		 * @return  The instance of the class to support the 'Fluent interface' aka method-chaining
+		 */
+		public function header(headLine:String):Output
+		{
+			return text("<font size=\"36\" face=\"Georgia\"><u>" + headLine + "</u></font>\n");
+		}
+
+		/**
+		 * Clears the output-buffer
+		 *
+		 * @param   hideMenuButtons   Set this to true to hide the menu-buttons (rarely used)
+		 * @return  The instance of the class to support the 'Fluent interface' aka method-chaining
+		 */
+		public function clear(hideMenuButtons:Boolean = false):Output
+		{
+			if (hideMenuButtons) {
+				forceUpdate();
+				if (kGAMECLASS.gameState != 3) mainView.hideMenuButton(MainView.MENU_DATA);
+				mainView.hideMenuButton(MainView.MENU_APPEARANCE);
+				mainView.hideMenuButton(MainView.MENU_LEVEL);
+				mainView.hideMenuButton(MainView.MENU_PERKS);
+				mainView.hideMenuButton(MainView.MENU_STATS);
+			}
+
+			_currentText = "";
+			mainView.clearOutputText();
+			return this;
+		}
+
+		/**
+		 * Adds raw text to the output without passing it through the parser
+		 *
+		 * If you want to clear the output before adding raw text, use clear(true) or just clear()
+		 * The second param `purgeText:Boolean = false` is not supported anymore in favor of using clear() and will never return.
+		 *
+		 * @param   text    The text to be added to the output-buffer
+		 * @return  The instance of the class to support the 'Fluent interface' aka method-chaining
+		 */
+		public function raw(text:String):Output
```

Comment:
Actually only one call to rawOutputText() in the whole codebase hides the buttons, too ([line 85 of includes/eventParser.as](https://github.com/Kitteh6660/Corruption-of-Champions-Mod/blob/master/includes/eventParser.as#L85) to be precise), so you should always use `output.clear();`, when you want to clear the output before adding raw text to it.